tests/test_writer.py

```diff
@@ -43,7 +43,7 @@ def test_format_requirement_annotation_case_sensitive(from_line, writer):
     assert (writer._format_requirement(ireq,
                                        reverse_dependencies,
                                        primary_packages=[]) ==
-            'Test==1.2                 ' + comment('# via xyz'))
```

Comment:
Yes that makes sense. 

Going forward, I think adding and maintaining tests is really important, particularly in this new "distributed" maintainership model (jazzband) that pip-tools is using. I know it is more work, but it is better to spend time locally than distribute the debugging effort on the internet :-). 

Maybe this should be included in the 1.8.1 release I am preparing.  A release candidate 1 is out already.